### PR TITLE
add ttl article

### PR DESCRIPTION
**Pull request by ran-huang**
Signed-off-by: Ran <huangran@pingcap.com>

### CODE REVIEW

File: managing-you-data-lifecycle-with-time-to-live-tables.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+![Row-granularity TTL table](media/ttl-table-by-row.gif)
+<div class="caption-center">Row-granularity TTL table</div>
+
+When TiDB dispatches GC tasks to TiKV, the tasks include the key ranges corresponding to the TTL tables and their TTL configuration. Based on the TTL configuration, TiKV calculates the lifetime of the data within the TTL key range. If TiKV finds data that is valid in MVCC but exceeds TTL, TiKV deletes the data during the GC process and releases the storage space.
+
+Notably, **most TiDB tables currently use [non-clustered indexes](https://docs.pingcap.com/tidb/stable/clustered-indexes)**. If the primary key index or secondary index has inconsistent deletion progress with the table data, a set of data that is already deleted from the table might still be visible in the index. This leads to query failure when any index is used. To solve this problem, we split the TTL table into two key ranges: data and index. `The TTL for data =  the TTL for index + the interval between the recent two rounds of GC`. In this way, all data lives longer than the index for at least one GC period.
```
Comment (CaitinChen) [inline managing-you-data-lifecycle-with-time-to-live-tables.md]:
```suggestion
Notably, **most TiDB tables currently use [non-clustered indexes](https://docs.pingcap.com/tidb/stable/clustered-indexes)**. If the primary key index or secondary index has inconsistent deletion progress with the table data, a set of data that is already deleted from the table might still be visible in the index. This leads to query failure when any index is used. To solve this problem, we split the TTL table into two key ranges: data and index. `The TTL for data = the TTL for index + the interval between the recent two rounds of GC`. In this way, all data lives longer than the index for at least one GC period.
```

File: managing-you-data-lifecycle-with-time-to-live-tables.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+During the Hackathon, we adapted a popular MQTT broker to make it support MySQL as backend storage. Because TiDB is MySQL compatible, **TiDB's TTL table can automatically delete the expired data according to the retention cycle set by the user**.
+
+## Summary
+
+In an era of everything connected with everything, the volume of data is rapidly growing. How to support more data with fewer costs is a question that all DBMSs try to answer. Our answer to this question is the TTL table.
+
+For TiDB, the heart of this question is how to balance the cost of TiDB and the value of data it carries. The more relevant and fresher data is, the more value TiDB creates for your business. After all, time waits for no one, and it's better to make room for more valuable data.
+
+_At TiDB Hackathon 2020, many excellent, interesting projects were born. We'll be telling you about them in future [blog posts](https://pingcap.com/blog/tag/Hackathon). Stay tuned._
```
Comment (CaitinChen) [inline managing-you-data-lifecycle-with-time-to-live-tables.md:119]: Here, before "TiDB Hackathon 2020," you use "at", but you used "in" in the preceding content. You can update "in TiDB Hackathon 2020" to "at TiDB Hackathon 2020" to ensure consistent language style.

File: managing-you-data-lifecycle-with-time-to-live-tables.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+**Author:** [Xiaoguang Sun](https://github.com/sunxiaoguang) (Software Architect at Zhihu, TiKV Project Maintainer)
+
+**Transcreator:** [Ran Huang](https://github.com/ran-huang); **Editor:** Tom Dewan
+
+![Manage data lifecycle with time to live tables in distributed SQL database](media/manage-data-lifecycle-with-time-to-live-tables-in-distributed-sql-database.jpg)
+
+Your organization is growing with each passing day; so is your data. More data brings more business opportunities, but it also begets higher storage costs. You want a better way to manage the cost? We want the same thing for our open source database, TiDB.
+
+[TiDB](https://docs.pingcap.com/tidb/stable) is a distributed SQL database designed for massive data. Our goal is to support large-scale datasets with a reasonable cost. In [TiDB Hackathon 2020](https://pingcap.com/community/events/hackathon2020/), we took a big step in that direction. **We introduced a feature, [the time to live (TTL) table](https://github.com/tidb-hackathon/t4), that enables TiDB to automatically manage the lifecycle of data according to its lifetime**. TiDB makes sure every portion of its resources is consumed by high-value, fresh data.
```
Comment (ran-huang) [inline managing-you-data-lifecycle-with-time-to-live-tables.md:19]:
```suggestion
[TiDB](https://docs.pingcap.com/tidb/stable) is a distributed SQL database designed for massive data. Our goal is to support large-scale datasets with a reasonable cost. At [TiDB Hackathon 2020](https://pingcap.com/community/events/hackathon2020/), we took a big step in that direction. **We introduced a feature, [the time to live (TTL) table](https://github.com/tidb-hackathon/t4), that enables TiDB to automatically manage the lifecycle of data according to its lifetime**. TiDB makes sure every portion of its resources is consumed by high-value, fresh data.
```

File: managing-you-data-lifecycle-with-time-to-live-tables.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+![Row-granularity TTL table](media/ttl-table-by-row.gif)
+<div class="caption-center">Row-granularity TTL table</div>
+
+When TiDB dispatches GC tasks to TiKV, the tasks include the key ranges corresponding to the TTL tables and their TTL configuration. Based on the TTL configuration, TiKV calculates the lifetime of the data within the TTL key range. If TiKV finds data that is valid in MVCC but exceeds TTL, TiKV deletes the data during the GC process and releases the storage space.
+
+Notably, **most TiDB tables currently use [non-clustered indexes](https://docs.pingcap.com/tidb/stable/clustered-indexes)**. If the primary key index or secondary index has inconsistent deletion progress with the record, a set of data that is already deleted from the table might still be visible in the index. This leads to query failure when any index is used. To solve this problem, we split the TTL table into two key ranges: data and index. `The TTL for data =  the TTL for index + the interval between the recent two rounds of GC`. In this way, all data lives longer than the index for at least one GC period.
```
Comment (ran-huang) [inline managing-you-data-lifecycle-with-time-to-live-tables.md:44]:
```suggestion
Notably, **most TiDB tables currently use [non-clustered indexes](https://docs.pingcap.com/tidb/stable/clustered-indexes)**. If the primary key index or secondary index has inconsistent deletion progress with the record, a set of data that is already deleted from the table might still be visible in the index. This leads to query failure when any index is used. To solve this problem, we split the TTL table into two key ranges: data and index. `The TTL for data = the TTL for index + the interval between the recent two rounds of GC`. In this way, all data lives longer than the index for at least one GC period.
```

File: managing-you-data-lifecycle-with-time-to-live-tables.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
+---
+title: Managing Your Data Lifecycle with Time to Live Tables
+author: ['Xiaoguang Sun']
+date: 2021-06-10
+summary: Time to live, or TTL, is a mechanism to manage the data lifecycle in a system. In TiDB Hackathon 2020, a team introduced TTL tables into TiDB to enable automatic management of the data lifecycle.
```
Comment (CaitinChen) [inline managing-you-data-lifecycle-with-time-to-live-tables.md:5]:
```suggestion
summary: Time to live, or TTL, is a mechanism to manage the data lifecycle in a system. At TiDB Hackathon 2020, a team introduced TTL tables into TiDB to enable automatic management of the data lifecycle.
```

**Review by CaitinChen — APPROVED** [review]
LGTM

**Pull request merged**